airbyte-integrations/connectors/destination-langchain/destination_langchain/document_processor.py

```diff
@@ -84,7 +84,7 @@ def _extract_metadata(self, record: AirbyteRecordMessage) -> dict:
         metadata[METADATA_STREAM_FIELD] = stream_identifier
         # if the sync mode is deduping, use the primary key to upsert existing records instead of appending new ones
         if current_stream.primary_key and current_stream.destination_sync_mode == DestinationSyncMode.append_dedup:
-            metadata[METADATA_RECORD_ID_FIELD] = self._extract_primary_key(record, current_stream)
+            metadata[METADATA_RECORD_ID_FIELD] = f"{stream_identifier}_{self._extract_primary_key(record, current_stream)}"
```

Comment:
Is this a breaking change? as in, will we correctly dedupe records if there are already records present that use the previous behavior?

This probably requires running a reset, right?

Reply:
This is alpha so we're free to make breaking changes (though we should probably bump to 0.1.0 if so) - mainly want to confirm my understanding

Reply:
Follow up question, how would you want to handle the rollout of this breaking change? We could use the new breaking change workflow to show messages in-app and instruct folks on what to do, but maybe the usage is low enough to the point we don't think this is necessary right now.